### ISSUE TITLE
'vagrant list' does not show the state of VMs for vmware_fusion provider

### ISSUE THREAD

**Issue by romanrev** [joshmcarthur/vagrant-list#5]
It seems that this plugin is only useful to list the virtualbox VMs only:

```
for i in $(vagrant box list | awk '{print $1}' ); do [ -x "$i" ] && ( echo "$i:"; cd $i; vagrant status; ); done
centos6.4_i686minPupChef:
Current machine states:

default                  saved (virtualbox)

To resume this VM, simply run 'vagrant up'.

centos6.4_x86_64_fusion:
Current machine states:

default                  running (vmware_fusion)

The VM is running. To stop this VM, you can run `vagrant halt` to
shut it down, or you can run `vagrant suspend` to simply suspend
the virtual machine. In either case, to restart it again, run
'vagrant up'.

ubuntu12.04_amd64_fusion_omnibus:
Current machine states:

default                  suspended (vmware_fusion)

The VM is suspended. To resume this VM, run 'vagrant up'. 
```

Here's the output of the 'vagrant list':

```
$ vagrant list
ALL:
baafeb09-f135-429b-8fe2-1b46b7a6f843: centos6.4_x86_64minPupChef_1366189724 (Red Hat)
RUNNING:
```

One can see that it provides the list of the virtualbox VMs only.

**Comment by joshmcarthur** [comment]
@romanrev yes, you're correct that it currently only works with VirtualBox VMs. The current strategy is to basically just parse the output of `VBoxManage list vms`. The script snippet you've posted above looks like it could be quite a nice replacement for this strategy though - do you mind if I use it to play around and see if I can come up with a cross-provider strategy to build into the gem?

**Comment by romanrev** [comment]
@joshmcarthur: Yes, that's what I found out looking inside the Gem later on :-) The snippet I pasted is very basic in fact and relies on the fact that the Vagrantfiles are placed inside the corresponding subdirectories, so I doubt it could be a good replacement. 
I tried to look in ~/.vagrant to find the ways how to figure out the boxes - no luck up to now. The VMware vmrun command is also not so helpful, as it can list only the running boxes, not all the defined ones. Besides I've stumbled into the issue of CentOS box not showing up in Fusion dashboard after initializing it with Vagrant (Ubuntu one is showing up fine though), so it _might_ be another rabbit hole.

**Comment by joshmcarthur** [comment]
@romanrev OK. Another option which occurred to me was that if `vagrant status` is able to print the status of the boxes, it must have a cross-provider method of finding this out, which the gem itself could leverage. Of course, that still leaves the problem of how to find the list of VMs that vagrant knows about, which might be hard, since as far as I know, vagrant doesn't share state with anything much.

And of course there's always the possibility of extending the naive approach I've got at the moment and just chaining some calls to all the provider utilities to list VMs - `vmrun`, `VBoxManage`, etc. ;-)

**Comment by beporter** [comment]
+1 / subscribing

**Comment by MuschPusch** [comment]
Since i missed to readt the vagrant 1.6 release notes: 

  $ vagrant global-status

makes vagrant list obsolete.
